components/raftstore/src/store/read_queue.rs

```diff
@@ -164,12 +164,17 @@ impl ReadIndexQueue {
     }
 
     /// A fast path for `advance_leader_reads` then `pop_front`.
-    pub fn advance_leader_read_and_pop(&mut self, uuid: Uuid, index: u64) -> ReadIndexRequest {
-        let mut read = self.reads.pop_front().unwrap();
-        self.handled_cnt += 1;
-        assert_eq!(uuid, read.id);
-        read.read_index = Some(index);
-        read
+    pub fn advance_leader_read_and_pop(&mut self, uuid: Uuid, index: u64) -> Vec<ReadIndexRequest> {
```

Comment:
Does `advance_leader_reads` have the same problem?

Reply:
No because `ready_cnt` is considered in the function. I add a new test case to check it.

Reply:
In most case, there should be only one entry. Is there a way to omit the allocation?